crowbar_framework/app/models/cinder_service.rb

```diff
@@ -68,11 +68,11 @@ def create_proposal
     base = super
 
     nodes = NodeObject.all
-    controller = select_nodes_for_role(nodes, "cinder-controller", "controller") || []
+    controllers = select_nodes_for_role(nodes, "cinder-controller", "controller") || []
     storage = select_nodes_for_role(nodes, "cinder-volume", "storage") || []
 
     base["deployment"][@bc_name]["elements"] = {
-      "cinder-controller" => [ controller.first.name ],
+      "cinder-controller" => controllers.empty? ? [] : [ controllers.first.name ],
```

Comment:
ok, with the select_nodes_for_role fix I'd tentatively prefer the .map approach..